trng/limits.hpp

```diff
@@ -38,7 +38,7 @@
 #include <cfloat>
 #include <trng/cuda.hpp>
 
-#if defined TRNG_CUDA
+#if defined TRNG_CUDA && !defined __HIPCC__
```

Comment:
In the context of adding HIP support, this should rather be 
```
#if defined __CUDACC__
```

Also the macro `TRNG_CUDA` changes is meaning, it' not longer CUDA related.  I tend to replace it by something else, e.g., `TRNG_DEVICE_ATTRIBUTE`.

Reply:
I agree with you. It should have been renamed TRNG_GPU_SUPPORT or something similar but I felt it was a major refactoring. Therefore before undertaking it I wanted to check with you.